python/ray/rllib/utils/tf_policy_graph.py

```diff
@@ -99,15 +111,30 @@ def compute_actions(
             builder, obs_batch, state_batches, is_training)
         return builder.get(fetches)
 
-    def _get_loss_inputs_dict(self, postprocessed_batch):
+    def _get_loss_inputs_dict(self, batch):
         feed_dict = {}
-        for key, ph in self._loss_inputs:
-            # TODO(ekl) fix up handling of RNN inputs so that we can batch
-            # across multiple rollouts
-            if key.startswith("state_in_"):
-                feed_dict[ph] = postprocessed_batch[key][:1]  # in state only
-            else:
-                feed_dict[ph] = postprocessed_batch[key]
+
+        # Simple case
+        if not self._state_inputs:
+            for k, ph in self._loss_inputs:
+                feed_dict[ph] = batch[k]
+            return feed_dict
+
+        # RNN case
+        feature_keys = [
+            k for k, v in self._loss_inputs if not k.startswith("state_in_")]
```

Comment:
should add note that if someone bakes their own RNN model and wants to use this functionality, they need to prefix like this

Reply:
These are automatically added by sampler, so it's not needed.